lib/ansible/modules/network/cloudengine/ce_snmp_traps.py

```diff
@@ -0,0 +1,552 @@
+                if self.cur_cfg["snmp-agent trap"]:
+                    for item in self.cur_cfg["snmp-agent trap"]:
+                        if item == tmp_cfg:
+                            find_flag = True
+                        elif item == "enable":
+                            find_flag = True
+                        elif tmp_cfg in item:
+                            find_flag = True
+                else:
+                    find_flag = True
+
+                find_undo_flag = False
+                if self.cur_cfg["undo snmp-agent trap"]:
+                    for item in self.cur_cfg["undo snmp-agent trap"]:
+                        if item == tmp_cfg:
+                            find_undo_flag = True
+                        elif tmp_cfg in item:
+                            find_undo_flag = True
+
+                if find_undo_flag:
+                    pass
+                elif find_flag:
+                    self.undo_trap_feature_name()
+
+            if self.interface_type:
+                if "trap source interface" in self.cur_cfg.keys():
+                    self.undo_trap_source_interface()
+
+            if self.port_number:
+                if "trap source-port" in self.cur_cfg.keys():
+                    self.undo_trap_source_port()
+
+        self.get_end_state()
+
+        self.results['changed'] = self.changed
+        self.results['proposed'] = self.proposed
+        self.results['existing'] = self.existing
+        self.results['end_state'] = self.end_state
+        self.results['updates'] = self.updates_cmd
+
+        self.module.exit_json(**self.results)
+
+
+def main():
+    """ Module main """
+
+    argument_spec = dict(
+        state=dict(choices=['present', 'absent'], default='present'),
+        feature_name=dict(choices=['aaa', 'arp', 'bfd', 'bgp', 'cfg', 'configuration', 'dad',
+                                   'devm', 'dhcpsnp', 'dldp', 'driver', 'efm', 'erps', 'error-down',
+                                   'fcoe', 'fei', 'fei_comm', 'fm', 'ifnet', 'info', 'ipsg', 'ipv6',
+                                   'isis', 'l3vpn', 'lacp', 'lcs', 'ldm', 'ldp', 'ldt', 'lldp',
+                                   'mpls_lspm', 'msdp', 'mstp', 'nd', 'netconf', 'nqa', 'nvo3',
+                                   'openflow', 'ospf', 'ospfv3', 'pim', 'pim-std', 'qos', 'radius',
+                                   'rm', 'rmon', 'securitytrap', 'smlktrap', 'snmp', 'ssh', 'stackmng',
+                                   'sysclock', 'sysom', 'system', 'tcp', 'telnet', 'trill', 'trunk',
+                                   'tty', 'vbst', 'vfs', 'virtual-perception', 'vrrp', 'vstm', 'all']),
+        trap_name=dict(type='str'),
+        interface_type=dict(choices=['Ethernet', 'Eth-Trunk', 'Tunnel', 'NULL', 'LoopBack', 'Vlanif',
+                                     '100GE', '40GE', 'MTunnel', '10GE', 'GE', 'MEth', 'Vbdif', 'Nve']),
+        interface_number=dict(type='str'),
+        port_number=dict(type='str')
```

Comment:
Because the device side is string, avoid repeated conversion,  retain it， thanks.